File: go-lab-5/task-1.py
import math


class Person:
    def __init__(self, name, age):
        self.name = name
        self.age = age

    def get_info(self):
        return f"Звать {self.name}, {self.age} лет"


def task1():
    print("\n>> Задача 1")

    print(person.get_info())


def task2():
    print("\n>> Задача 2")
    print("С днюшечкой!!")
    person.age += 1
    print(person.get_info())


class Circle:
    def __init__(self, radius):
        self.radius = radius

    def area(self):
        return self.radius * self.radius * math.pi


def task3():
    print("\n>> Задача 3")
    print(f"Площадь круга: {circle.area():.2f}")


class Rectangle:
    def __init__(self, width, height):
        self.width = width
        self.height = height

    def area(self):
        return self.width * self.height


def task4():
    print("\n>> Задача 4")
    print("Площади:")
    for shape in shapes:
        print(f"{shape.area():.2f}")


def slice_shapes(shapes):
    for shape in shapes:
        print(f"{shape.area():.2f}")


def task5():
    print("\n>> Задача 5")
    print("Те же площади, но функцией:")
    slice_shapes(shapes)


class Book:
    def __init__(self, title, author, genre):
        self.title = title
        self.author = author
        self.genre = genre

    def get_info(self):
        return f"«{self.title}» от автора {self.author} ({self.genre})"


def task6():
    print("\n>> Задача 6")
    print("Книжная инфа:")

    print(book.get_info())


def main():
    global person, circle, shapes, book

    person = Person("Артем", 20)

    circle = Circle(12.5)

    shapes = [
        Circle(17),
        Rectangle(20.2, 7)
    ]

    book = Book("Война и мир", "Лев Николаевич Толстой", "Роман")

    task1()
    task2()
    task3()
    task4()
    task5()
    task6()


if __name__ == "__main__":
    main()

File: go-lab-5/task-1.go
package main

import (
	f "fmt"
	m "math"
)

func (person *Person) getInfo() string {
	return f.Sprintf("Звать %s, %d лет", person.name, person.age)
}

type Person struct {
	name string
	age  int
}

func task1() {
	f.Println("\n>> Задача 1")

	f.Println(person.getInfo())
}

//

func (person *Person) birthday() {
	person.age++
}

func task2() {
	f.Println("\n>> Задача 2")

	f.Println("С днюшечкой!!")
	person.birthday()

	f.Println(person.getInfo())
}

//

func (circle *Circle) Area() float64 {
	return circle.radius * circle.radius * m.Pi
}

type Circle struct {
	radius float64
}

func task3() {
	f.Println("\n>> Задача 3")

	f.Printf("Площадь круга: %.2f\n", circle.Area())
}

//

func (r *Rectangle) Area() float64 {
	return r.width * r.height
}

type Rectangle struct {
	width, height float64
}

type Shape interface {
	Area() float64
}

func task4() {
	f.Println("\n>> Задача 4")

	f.Println("Площади:")
	for _, shape := range shapes {
		f.Printf("%.2f\n", shape.Area())
	}
}

func sliceShapes(shapes []Shape) {
	for _, shape := range shapes {
		f.Printf("%.2f\n", shape.Area())
	}
}

func task5() {
	f.Println("\n>> Задача 5")

	f.Println("Те же площади, но функцией:")
	sliceShapes(shapes)
}

//

func (book *Book) getInfo() string {
	return f.Sprintf("«‎%s» от автора %s (%s)",
		book.title, book.author, book.genre)
}

type Stringer interface {
	getInfo() string
}

type Book struct {
	title  string
	author string
	genre  string
}

func task6() {
	f.Println("\n>> Задача 6")

	f.Println("Книжная инфа:")
	book := Book{
		title:  "Война и мир",
		author: "Лев Николаевич Толстой",
		genre:  "Роман",
	}
	f.Println(book.getInfo())
}

//

var person = Person{
	name: "Артем",
	age:  20,
}

var circle = Circle{
	radius: 12.5,
}

var shapes = []Shape{
	&Circle{radius: 17},
	&Rectangle{width: 20.2, height: 7}}

func main() {
	task1()
	task2()
	task3()
	task4()
	task5()
	task6()

}
